### PR TITLE
remove unnecessary 'setuptools' and 'wheel' dependencies

### PR DESCRIPTION
**Pull request by jameslamb**
Contributes to https://github.com/rapidsai/build-planning/issues/62.

It looks like the `cugraph` and `pylibcugraph` wheels and conda recipes have unnecessary dependencies on `setuptools` and `wheel`. I suspect those are left over from before those libraries were cut over to `scikit-build-core`.

This proposes removing those.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

## Notes for Reviewers

How I confirmed there were no direct uses of `setuptools` in `cugraph` and `pylibcugraph`:

```shell
git grep -i setuptools
```

### CODE REVIEW

**Review by AyodeAwe — APPROVED** [review]

**Comment by jameslamb** [comment]
> `/merge`

Oh my bad, missed that this needs a review from a Python codeowner.

**Pull request merged**
